src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,103 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+import { getTodos, updateTodo, addTodo } from '../../api/todos';
+import { User } from '../../types/User';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>,
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setIsToggleClicked: React.Dispatch<React.SetStateAction<boolean>>,
+  user: User | null,
+  setIsAdding: React.Dispatch<React.SetStateAction<boolean>>,
+  isAdding: boolean,
+  setNotification: React.Dispatch<React.SetStateAction<string>>,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  newTodoField,
+  todos,
+  setTodos,
+  setIsToggleClicked,
+  user,
+  setIsAdding,
+  isAdding,
+  setNotification,
+}) => {
+  const [title, setNewTodoFieldValue] = useState('');
+
+  const handleToggler = async () => {
+    todos.forEach(async todo => {
+      setIsToggleClicked(true);
+      await updateTodo(todo.id, {
+        completed: !todo.completed,
+      });
+      await getTodos(todo.userId).then(setTodos);
+      setIsToggleClicked(false);
+    });
+  };
+
+  const handleSubmit = async () => {
+    if (title) {
+      const newTodo = {
+        id: 0,
+        userId: user ? user.id : 0,
```

Comment:
```suggestion
        userId: user?.id ?? 0,
```
Поідеї так тоже буде працювати, але це міняти необов'язково, це більше показати, шо таке існує, можеш почитати про nullish coalescing operator)